airflow/sensors/base.py

```diff
@@ -239,7 +211,8 @@ def poke(self, context: Context) -> bool | PokeReturnValue:
         """Override when deriving this class."""
         raise AirflowException("Override me.")
 
-    def execute(self, context: Context) -> Any:
+    @provide_session
+    def execute(self, context: Context, session=NEW_SESSION) -> Any:
```

Comment:
adding additional arg `session` is causing lot of mypy failures in providers where `BaseSensorOperator` is being extended.  Error's something like this:

```
providers/src/airflow/providers/microsoft/azure/sensors/data_factory.py:98: error:
Signature of "execute" incompatible with supertype "BaseSensorOperator" 
[override]
        def execute(self, context: Context) -> None:
        ^
providers/src/airflow/providers/microsoft/azure/sensors/data_factory.py:98: note:      Superclass:
providers/src/airflow/providers/microsoft/azure/sensors/data_factory.py:98: note:          def execute(context: Context, session: Any = ...) -> Any
providers/src/airflow/providers/microsoft/azure/sensors/data_factory.py:98: note:      Subclass:
providers/src/airflow/providers/microsoft/azure/sensors/data_factory.py:98: note:          def execute(self, context: Context) -> None
```

Should we just remove the decorator instead of updating execute method signature here?

Failing checks: https://github.com/apache/airflow/actions/runs/12101035549/job/33740515392?pr=44510

Reply:
I reverted this in https://github.com/apache/airflow/pull/44510

Reply:
Uuups, oh was at-sleep while the error was on main. If we revert I assume we should not make this is a batch with another PR - should we not "just" revert the PR from me that caused it?

Reply:
I just fixed it and it passed all tests. That should be good? I removed the decorator as well